### PR TITLE
 fix(button): fix theme tokens and focus style

### LINKED ISSUE #3190: [Button] Dark themes disabled state wrong & icon btn has extra border
### 1. Disabled states showing wrong color & hover state

- [ ] Disabled button should use `$disabled-02` for background and `$disabled-03` for text & icon. 

**Current:** 

Jun-26-2019 14-30-32

image

image

**Should be**

`$disabled-02` for background and `$disabled-03` for text & icon.

### 2. Icon button has double border on focus

- [ ] Inside focus border around icon itself should not be there. Only focus border around the button needed.

image

### PR DESCRIPTION
**Pull request by asudoh**
Fixes #3190.

#### Changelog

**Changed**

- Background color of disabled button.
- Text/icon colors of ghost button.

**Removed**

- Focus outline of `<svg>` in icon-only button.

#### Testing / Reviewing

Testing should make sure button styles are not broken.

### CODE REVIEW

**Review by joshblack — APPROVED** [review]

**Review by emyarod — APPROVED** [review]
looks good to me!

**Review by aagonzales — COMMENTED** [review]
image

Icon in the disabled button should also be `$disabled-03`* same as the text color.

File: packages/components/src/components/button/_button.scss [hunk 1]
```diff
@@ -118,9 +118,9 @@
     @include button-theme(
       transparent,
       transparent,
-      $interactive-04,
+      $icon-01,
```
Comment (aagonzales) [inline packages/components/src/components/button/_button.scss]: Also I wrote the wrong token in the initial [comment](https://github.com/carbon-design-system/carbon/issues/3190#issuecomment-506765800). Ghost button should be using `$link-01` for text + icon (not `$icon-01`). My bad guys.

Comment (aagonzales) [inline packages/components/src/components/button/_button.scss]: Also I wrote the wrong token in the initial [comment](https://github.com/carbon-design-system/carbon/issues/3190#issuecomment-506765800). Ghost button should be using `$link-01` for text + icon (not `$icon-01`). My bad guys.

**Comment by elizabethsjudd** [comment]
FYI, i created an issue (https://github.com/carbon-design-system/carbon/issues/3092) previously about the icon color so if @asudoh is going to address that in this PR, that issue can be resolved with the PR as well.

File: packages/components/src/components/button/_button.scss (hunk 1 above)
Comment (joshblack) [reply packages/components/src/components/button/_button.scss]: Just updated @aagonzales!

**Review by tw15egan — APPROVED** [review]
Looks good once @aagonzales signs off 👍

File: packages/components/src/components/button/_button.scss [hunk 2]
```diff
@@ -118,9 +118,9 @@
     @include button-theme(
       transparent,
       transparent,
-      $interactive-04,
+      $link-01,
       $hover-ui,
-      $interactive-04,
+      $icon-01,
```
Comment (aagonzales) [inline packages/components/src/components/button/_button.scss]: the icon also needs to be `link-01` sorry that wasn't clear.
image

**Review by aagonzales — COMMENTED** [review]
Also the icon on disabled button still isn't using the correct token, should be using `$disabled-03`
image

**Comment by asudoh** [comment]
@aagonzales Thank you for your additional review. Updated icon color tokens.

**Comment by jeanservaas** [comment]
Ghost button looks good, but we're still having the same issue with the disabled icon:

I can see the change from $ui-04 to $disabled-03 and $disabled-03 is obviously working for the text... is there some override on the icon? It's pulling the wrong value -- if you look at it, the disabled icon is the same color as the enabled icon.

image

**Comment by joshblack** [comment]
Looks like we have stuff from the primary button overriding the `path`:

image

**Comment by vpicone** [comment]
Is there any reason we're not just using `currentColor` for the icon fill? It's ie11 compliant and simplifies a lot of this. Example:

https://carbon-foundry-vpicone.tagteam1.now.sh/?path=/story/button--with-icon

**Comment by tw15egan** [comment]
I'm pretty sure we were previously using `currentColor`, so not sure if it just got lost in the migration but it seems like the easiest way to handle this

**Comment by joshblack** [comment]
I'm definitely down for this too 👍 Maybe we should plan it for next release in case we don't catch something when changing it now?

**Comment by elizabethsjudd** [comment]
@vpicone  FYI, `currentColor` can't animate. We used to use it in WH all the time and then when we started adding animations/transitions in the CSS we noticed the icon didn't respect the duration/easing (but it did change color).

**Comment by tw15egan** [comment]
@elizabethsjudd do you have an example? I just threw together a CodePen and I'm not seeing any immediate issues (although I could be misinterpreting the issues you had!) 

https://codepen.io/tjegan/pen/dBrgxb

**Comment by elizabethsjudd** [comment]
@tw15egan If i'm remembering right it was limited to certain browsers and this was probably about a year ago.  I just ran your code pen in Chrome, FF, Safari, and Mac Edge and had no problem so maybe support has been fixed!! (wooo for constantly changing technology) . I can double check on actual Edge and IE11 tomorrow when I have my windows laptop.

**Comment by tw15egan** [comment]
Woo! 🎉 That would be great if it's been fixed, although I'm not going to hold my breath on IE11 support...

**Comment by elizabethsjudd** [comment]
@tw15egan honestly i feel like this is a strong case for grace degradation for IE11.  I know the color changes it was just abrupt when we were testing it with the transitions/animations.

**Comment by tw15egan** [comment]
Yeah, as long as the color still changes I think we're fine with using `currentColor` and just (possibly) losing animations in IE11, as the pros seem to outweigh the cons

**Comment by asudoh** [comment]
Thank you for spotting @jeanservaas - Fixed the disabled icon color.

**Review by jeanservaas — APPROVED** [review]
LGTM!

**Pull request merged**